### DISCUSSION TITLE
Is there a good getting started example that showcase decorators, emitters, and library?

### DISCUSSION THREAD

**Discussion by akoserwal** [microsoft/typespec#3551] [Q&A]
I can't find a good example which I can compile locally and use as a base project to get started

**Comment by timotheeguerin** [comment]
For docs we just have step by steps instructions here https://typespec.io/docs/extending-typespec/basics

But we do have templates you can use to init an emitter or library
```
tsp init --template emitter-ts
tsp init --template library-ts
```

You can see the rendered template [here](https://github.com/microsoft/typespec/tree/main/packages/compiler/templates/__snapshots__)

**Reply by akoserwal** [reply]
Thank you for the instructions. This helps set up a basic project.

It would be good to add examples/library references.

I can clone a simple-library project and use it as a base reference. That will navigate the concepts faster.

Also I would be interested in some complex scenarios for the emitter/decorator, like

I am looking for examples of libraries/projects like https://github.com/Azure/typespec-azure
which helps understand complex cases like writing an emitter: https://github.com/Azure/typespec-azure/blob/main/packages/typespec-autorest-canonical/src/emitter.ts

**Reply by timotheeguerin** [reply]
ah yeah sorry, then yeah looking at our existing emitter/libraries is probably the best way today.
- xml - simpler with many decorators
- openapi3 or azure-tools/typespec-autorest quite complex emitter
- json schema decently complex emitter
- http - quite complex library

**Reply by akoserwal** [reply]
@timotheeguerin I am happy to collaborate and contribute to the creation of some example library projects. Where would be the best place to publish them so they are reachable?

**Reply by bterlson** [reply]
That's awesome to hear! More examples would be wonderful.

I would say you can put them on your github for now. We need to discuss what the best way to offer these examples are, e.g. whether we have a curated set of examples in this repo, or link to samples from the community, or some combination. Let me know if you have any opinion on that!

**Reply by akoserwal** [reply]
I'll try to create a repo and try to add a few examples. I will share the repo here to get some inputs.
